_RFCs/0004-testing-pull-requests.md

```diff
@@ -0,0 +1,47 @@
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+When someone opens a pull request to the core plans repo, they must include steps to test the change they are making (or the functionality of a new plan). The intent is to have a way to test any new functionality and, as much as possible, prevent breaking of existing functionality. Providing these testing steps (and sample apps when useful) is the responsibility of the person opening the pull request. However, should the person opening the pull request need help with establishing testing steps, they are most welcome to ask for it in the pull request.  A pull request must not be approved until testing steps are provided in the pull request.
+
+At least one core maintainer must review the pull request's code and, upon approval, either that same maintainer or another maintainer must follow the manual testing steps locally. This manual testing can be done either before or after the merge (when the package is on the unstable channel in Builder), but must be done before the package is promoted to stable. It should be noted that it is a bad practice to leave a package in unstable more than a day without testing it for promotion to stable (ideally, it should be tested as soon as it builds successfully and is put in the unstable channel).
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+Here are some examples of manually testing pull requests:
+
+* [Reviewing a binary package](https://forums.habitat.sh/t/reviewing-core-plans-example-binaries/449)
+* [Reviewing changes to core/scaffolding-node and core/scaffolding-ruby](https://forums.habitat.sh/t/reviewing-core-plans-example-scaffoldings/450)
+* [General guidelines for reviewing core plans](https://forums.habitat.sh/t/reviewing-core-plans/434)
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+This does put more of a burden on the person who is opening the pull request - to not only submit the request, but also include steps for testing it.
+
+# Rationale and alternatives
+[alternatives]: #alternatives
+
+There is an automated testing framework in the works, which will be very helpful with this. That said, as of the time of writing this pull request, that framework is not yet ready. Additionally, even when that framework is ready, we still need to know how to manually test a plan before we can automate those tests. The best person to guide how to test a change is the person who made that change.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+How will we handling currently open pull requests?
```

Comment:
Another question for now is: How do we manage packages that are used by many others? Should we re-test all of them?

Reply:
I don't think we need to go back and retest packages that are already in use at this time - but should commit to testing them from this time forward.

Reply:
That would mean a really awful lot of work for core maintainers for some packages (think glibc, or things like jre, node, ruby etc). Even curl is used by a lot of packages

Reply:
That's a good point.  The testing framework that @eeyun and I are working on will take care of testing that dependencies build, running defined tests on them, etc.  So help on that is coming!  But I think it is out of the scope of this particular RFC.

Reply:
In my opinion, we should work with the people that have currently open pull requests to get them tested, but since these rules weren't in place when those PRs were opened, they shouldn't be held to this new standard.

Reply:
Of course!  This standard would apply to all future pull requests.

Reply:
@raskchanky thank you for bringing that up, it is good to be specific about it.  I will remove that question from the RFC.